services/cases-api/helpers/schema.js

```diff
@@ -17,22 +17,39 @@ const caseAnswers = Joi.array().items(
   })
 );
 
+const encryptedAnswers = Joi.object({
+  encryptedAnswers: Joi.string(),
+});
+
+const formCurrentPosition = Joi.object({
+  index: Joi.number().required(),
+  level: Joi.number().required(),
+  currentMainStep: Joi.number().required(),
+  currentMainStepIndex: Joi.number().required(),
+});
+
+const signature = Joi.object({
+  success: Joi.bool().required(),
+});
+
 const form = Joi.object({
-  answers: caseAnswers.allow(),
-  currentPosition: Joi.object({
-    index: Joi.number().required(),
-    level: Joi.number().required(),
-    currentMainStep: Joi.number().required(),
-    currentMainStepIndex: Joi.number().required(),
-  }).required(),
+  answers: answers.allow(),
+  currentPosition: formCurrentPosition.required(),
 });
 
 const caseValidationSchema = Joi.object({
-  statusType: Joi.string().valid('notStarted').required(),
+  statusType: Joi.string().valid('notStarted'),
   currentFormId: uuid.required(),
   provider: caseProvider.required(),
   details: Joi.object().allow(),
   forms: Joi.object().pattern(/^/, [uuid.required(), form.required()]),
 });
 
+export const updateCaseValidationSchema = Joi.object({
+  currentFormId: uuid.required(),
+  answers: [answers.allow(), encryptedAnswers.allow()],
+  currentPosition: formCurrentPosition.required(),
+  signature: signature.optional(),
```

Comment:
Missing `provider` and `details`.

Reply:
Good point, then my next question are if these attributes are passed in any request from the app/client? otherwise it might be better to remove the option to update these values :)

Reply:
Good counter-point, another solution would then be to remove `provider` and `details` from `updateCase` 👍 .